File: backend/internal/model/incident.go
package model

import "time"

type Category string
type Status string

const (
    CategorySafety      Category = "Safety"
    CategoryMaintenance Category = "Maintenance"
)

const (
    StatusOpen       Status = "Open"
    StatusInProgress Status = "In Progress"
    StatusSuccess    Status = "Success"
)

type Incident struct {
    ID          string    `json:"id"          gorm:"primaryKey;type:uuid"`
    Title       string    `json:"title"       gorm:"not null"`
    Description string    `json:"description" gorm:"not null"`
    Category    Category  `json:"category"    gorm:"type:varchar(20);not null"`
    Status      Status    `json:"status"      gorm:"type:varchar(20);not null;default:'Open'"`
    CreatedAt   time.Time `json:"created_at"`
    UpdatedAt   time.Time `json:"updated_at"`
}

File: frontend/src/types/incident.ts
export type Category = 'Safety' | 'Maintenance';
export type Status = 'Open' | 'In Progress' | 'Success';

export interface Incident {
  id: string;
  title: string;
  description: string;
  category: Category;
  status: Status;
  created_at: string;
  updated_at: string;
}

export interface CreateIncidentDTO {
  title: string;
  description: string;
  category: Category;
  status: Status;
}

export interface UpdateIncidentDTO {
  title: string;
  description: string;
  category: Category;
  status: Status;
}


export interface IncidentQueryParams {
  category?: string;
  status?: string;
  search?: string;
  sort_by?: 'created_at' | 'updated_at' | 'title';
  sort_order?: 'asc' | 'desc';
  page?: number;
  page_size?: number;
}


export interface PaginatedResponse<T> {
  data: T[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}